### PR TITLE
fix(theme-default): sidebar scrollbar height error on ios safari and mask is invalid

### LINKED ISSUE #1103: [Bug]: sidebar scrollbar error and mask is invalid when using ios safari
### Version

```sh
System: iOS 
Browsers: Safari
```

### Details

When opening the sidebar, the document content can still be scrolled. In addition, there is a problem with the scrollbar area in the sidebar.
IMG_4991
IMG_4993

### Reproduce link

https://rspress.dev/

### Reproduce Steps

1. use ios safari
2. open [`rspress.dev`](https://rspress.dev)

### PR DESCRIPTION
**Pull request by shulaoda**
Fixed #1103 

## Related Issue
#1103 

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: packages/theme-default/src/components/Sidebar/index.tsx [hunk 1]
```diff
@@ -72,14 +77,26 @@ export let matchCache: WeakMap<
...
-  const [sidebarData, setSidebarData] = useState<
-    (ISidebarDivider | ISidebarItem | NormalizedSidebarGroup)[]
-  >(rawSidebarData.filter(Boolean).flat());
   const pathname = decodeURIComponent(rawPathname);
+  const langRoutePrefix = normalizeSlash(localesData.langRoutePrefix || '');
+
+  useEffect(() => {
+    if (inBrowser) {
+      if (isSidebarOpen) {
+        document.body.style.overflow = 'hidden';
```
Comment (Timeless0911) [inline packages/theme-default/src/components/Sidebar/index.tsx:94]: is this body style change have side effects

File: packages/theme-default/src/components/Sidebar/index.tsx [hunk 2]
```diff
@@ -186,27 +203,24 @@ export function Sidebar(props: Props) {
...
-          </div>
+      {!uiSwitch.showNavbar ? null : (
+        <div className={styles.navTitleMask}>
+          <NavBarTitle />
         </div>
+      )}
+      <div
+        className="rspress-scrollbar"
+        style={{
+          maxHeight: 'calc(100% - var(--rp-nav-height))',
```
Comment (Timeless0911) [inline packages/theme-default/src/components/Sidebar/index.tsx:214]: use css module?

File: packages/theme-default/src/components/Sidebar/index.tsx (hunk 1 above)
Comment (shulaoda) [reply packages/theme-default/src/components/Sidebar/index.tsx:94]: I referred to the implementation of `vitepress`, but I did not handle the special case where the overflow on the body may have an initial value.

Comment (Timeless0911) [reply packages/theme-default/src/components/Sidebar/index.tsx:94]: Every modification of the body style always brings some issues.... We need to be more cautious. Can you verify it more or can we not use the body style?

Comment (shulaoda) [reply packages/theme-default/src/components/Sidebar/index.tsx:94]: I think this is the best way, and its impact will only exist when the sidebar is opened.

Comment (shulaoda) [reply packages/theme-default/src/components/Sidebar/index.tsx:97]: There are some special circumstances that need to be considered and resolved.

File: packages/theme-default/src/components/Sidebar/index.module.scss [hunk 3]
```diff
@@ -21,12 +21,10 @@
   background: var(--rp-c-bg);
 }
 
-.sidebarContainer {
-  padding: 0 0 96px 0;
-}
-
-.sidebarContent {
+.sidebar-content {
```
Comment (Timeless0911) [inline packages/theme-default/src/components/Sidebar/index.module.scss:24]: camelcase is better to be consistent with other modules？

**Review by Timeless0911 — APPROVED** [review]

File: packages/theme-default/src/components/Sidebar/index.tsx (hunk 1 above)
Comment (Timeless0911) [reply packages/theme-default/src/components/Sidebar/index.tsx:97]: nothing yet

**Comment by chenjiahan** [comment]
Thanks! ❤️

**Pull request merged**